src/Property.php

```diff
@@ -96,6 +100,16 @@ public function setNullable(bool $bool): void
         $this->isNullable = $bool;
     }
 
+    public function isImmutable(): bool
+    {
+        return $this->isImmutable;
+    }
+
+    public function setIsImmutable(bool $isImmutable): void
```

Comment:
Why do we need a setter? The variable should be set from inside the `resolveTypeDefinition` call, so we can directly call it.

Reply:
for external packages who want control of things in the property class :)